Common/Utils/include/CommonUtils/ConfigurableParam.h

```diff
@@ -194,10 +189,12 @@ class ConfigurableParam
   template <typename T>
   static T getValueAs(std::string key)
   {
-    if (!sIsFullyInitialized) {
-      initialize();
-    }
-    return sPtree->get<T>(key);
+    return [](auto* tree, const std::string& key) -> T {
+      if (!sIsFullyInitialized) {
+        initialize();
+      }
+      return tree->template get<T>(key);
+    }(sPtree, key);
```

Comment:
what's the advantage of this over the original version?

Reply:
Unrelated changes now dropped.